### PR TITLE
WCMS-982 support for image proxy

### PR DESCRIPTION
**Pull request by mheiniger**
https://tracking.nzzmg.ch/browse/WCMS-982

**This is a BC break**: instead of passing config values on execution of the rendering, i pass the config object in the constructor of the RenderDecorator.

* re-add height to images
* don't remove imageratio anymore
* pass config object
* fix test and fixtures
* update readme

### CODE REVIEW

File: src/renderer/mobile-api/index.coffee [hunk 1]
```diff
@@ -56,10 +56,16 @@ removeHeader = (livingdoc) ->
   header.remove() if header?
 
 
-moveHeaderImageToLeadImage = (livingdoc, metadata, result) ->
+setLeadImage = (livingdoc, params, result) ->
+  metadata = params.metadata
+  leadImage = metadata.teaserImage
```
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: i would write a comment here, that we decided to take the teaser image as lead image.

Comment (mheiniger) [inline src/renderer/mobile-api/index.coffee]: a comment here

File: src/renderer/mobile-api/index.coffee [hunk 2]
```diff
@@ -56,10 +56,16 @@ removeHeader = (livingdoc) ->
...
+  if leadImage?.crops
+    for crops in leadImage.crops
```
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: for collections i would usually use a more functional approach. I tried if it works for this case and it looks like "clever code" :-(

```
leadImage.url = _.first(_.filter crops, (crop) -> crop.name == "16:9")?.url
```

@marcbachmann what do you think?

File: src/renderer/mobile-api/index.coffee [hunk 3]
```diff
@@ -56,10 +56,16 @@ removeHeader = (livingdoc) ->
...
+      leadImage.url = crops.url if crops.name == "16:9"
+
   headerImage = livingdoc.componentTree.find('image')[0]
-  result.json.leadImage = handler.handleImageComponent({ content : { image: metadata.teaserImage, caption: metadata.teaserCaption } }, metadata).boxes[0] if metadata.teaserImage
-  result.json.leadImage = handler.handleImageComponent(headerImage.toJson(), metadata).boxes[0] if headerImage
+  result.json.leadImage = handler.handleImageComponent({ content : { image: leadImage, caption: metadata.teaserCaption } }, params).boxes[0] if leadImage
```
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: this lines are too long and not readable anymore

File: src/renderer/mobile-api/index.coffee [hunk 4]
```diff
@@ -56,10 +56,16 @@ removeHeader = (livingdoc) ->
...
+  result.json.leadImage = handler.handleImageComponent(headerImage.toJson(), params).boxes[0] if not leadImage and headerImage
```
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: are you sure that boxes[0] exists?

File: test/renderer/mobile-api-renderer.coffee [hunk 5]
```diff
@@ -96,19 +111,15 @@ testMobileRenderer = (t, inputFileName, outputFileName, designNumber, options =
...
 
   # Run renderer
   document_id = fixture.document.revision.document_id || fixture.document.id
-  result = mobileApiRenderer.render(data, document_id,
-    webHost: "http://www.homer.nine.nzz.ch"
-  ).json
+  result = mobileApiRenderer.render(data, document_id).json
 #  console.log JSON.stringify(expected)
-#  console.log JSON.stringify(result)
+#  console.log JSON.stringify(result, null, 2)
```
Comment (DaRaFF) [inline test/renderer/mobile-api-renderer.coffee]: pls remove commented sections

File: src/renderer/mobile-api/componentHandler.coffee [hunk 6]
```diff
@@ -288,14 +289,18 @@ parseIframe = (match) ->
   if refsInline then formatReference('iframe', parsed) else ''
 
 
-createImage = (data) ->
+createImage = (data, params) ->
   if data.imageService = 'resrc.it'
     resrcItPath = data.url.replace(data.originalUrl, '') # temporarily remove image url
-    resrcItPath = resrcItPath.replace(/\/C=AR[\d.]+x[\d.]+/, '') # remove fixed ratio
+#    resrcItPath = resrcItPath.replace(/\/C=AR[\d.]+x[\d.]+/, '') # remove fixed ratio
```
Comment (DaRaFF) [inline src/renderer/mobile-api/componentHandler.coffee]: pls remove commented sections

**Comment by DaRaFF** [comment]
review done

File: src/renderer/mobile-api/index.coffee (hunk 1 above)
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: ok :+1:

File: test/renderer/mobile-api-renderer.coffee (hunk 5 above)
Comment (mheiniger) [inline test/renderer/mobile-api-renderer.coffee]: I use those lines every time i write new tests, what do you suggest to do, so i don't have to write it by hand every time?

Comment (DaRaFF) [inline test/renderer/mobile-api-renderer.coffee]: bö. dont know. If you really need it, leave it.

File: src/renderer/mobile-api/index.coffee (hunk 2 above)
Comment (DaRaFF) [inline src/renderer/mobile-api/index.coffee]: öh, i found an even better readable approach :-)

     crops.link = (_.find crops, (crop) -> crop.name == "16:9")?.link

Comment (marcbachmann) [inline src/renderer/mobile-api/index.coffee]: `_.find(crops, name: '16:9')`

File: src/renderer/mobile-api/index.coffee (hunk 4 above)
Comment (mheiniger) [inline src/renderer/mobile-api/index.coffee]: yes

**Pull request merged**